File: ruby/flatten/flatten.rb
class Array
  def flatten(level = -1)
    init = []
    each do |a|
      if a.is_a?(Array) && !level.zero?
        level -= 1
        init +=  a.flatten(level)
      else
        init << a
      end
    end
    init
  end
end

File: javascript/flatten/flatten.js
module.exports = function Flatten(array = [], level = -1, flatArray = []) {
  if (!array.length) { return array }

  array.forEach(function(item) {
    if ( Array.isArray(item) && level != 0 ) {
      level--
      Flatten(item, level, flatArray)
    } else {
      flatArray.push(item)
    }
  })

  return flatArray
}
